src/main/java/com/google/devtools/build/lib/unix/ProcMeminfoParser.java

```diff
@@ -54,11 +56,14 @@ public ProcMeminfoParser(String fileName) throws IOException {
       String valString = line.substring(colon + 1);
       try {
         long val =  Long.parseLong(CharMatcher.inRange('0', '9').retainFrom(valString));
-        builder.put(keyword, val);
+        if (!newMemInfo.containsKey(keyword)) {
+          newMemInfo.put(keyword, val);
+        }
       } catch (NumberFormatException e) {
         // Ignore: we'll fail later if somebody tries to capture this value.
       }
     }
+    builder.putAll(newMemInfo);
```

Comment:
Just change this into `memInfo = ImmutableMap.copyOf(newMemInfo);`, and drop the builder variable above. No need to create three maps if two are enough.

Reply:
good point, changed